ext/node/ops/os.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::error::AnyError;
+use deno_core::op;
+use deno_core::OpState;
+use errno::errno;
+use errno::set_errno;
+use errno::Errno;
+
+#[op]
+pub fn op_node_os_get_priority<P>(
+  state: &mut OpState,
+  pid: u32,
+) -> Result<i32, AnyError>
+where
+  P: NodePermissions + 'static,
+{
+  {
+    let permissions = state.borrow_mut::<P>();
+    permissions.check_sys("getPriority", "node:os.getPriority()")?;
+  }
+
+  get_priority(pid)
+}
+
+#[op]
+pub fn op_node_os_set_priority<P>(
+  state: &mut OpState,
+  pid: u32,
+  priority: i32,
+) -> Result<(), AnyError>
+where
+  P: NodePermissions + 'static,
+{
+  {
+    let permissions = state.borrow_mut::<P>();
+    permissions.check_sys("setPriority", "node:os.setPriority()")?;
+  }
+
+  set_priority(pid, priority)
+}
+
+#[op]
+pub fn op_node_os_username<P>(state: &mut OpState) -> Result<String, AnyError>
+where
+  P: NodePermissions + 'static,
+{
+  {
+    let permissions = state.borrow_mut::<P>();
+    permissions.check_sys("userInfo", "node:os.userInfo()")?;
+  }
+
+  Ok(whoami::username())
+}
+
+use crate::NodePermissions;
+#[cfg(unix)]
+use libc::id_t;
+#[cfg(unix)]
+use libc::PRIO_PROCESS;
+#[cfg(windows)]
+use winapi::shared::minwindef::DWORD;
+#[cfg(windows)]
+use winapi::shared::minwindef::FALSE;
+#[cfg(windows)]
+use winapi::shared::ntdef::NULL;
+#[cfg(windows)]
+use winapi::um::handleapi::CloseHandle;
+#[cfg(windows)]
+use winapi::um::processthreadsapi::GetCurrentProcess;
+#[cfg(windows)]
+use winapi::um::processthreadsapi::GetPriorityClass;
+#[cfg(windows)]
+use winapi::um::processthreadsapi::OpenProcess;
+#[cfg(windows)]
+use winapi::um::processthreadsapi::SetPriorityClass;
+#[cfg(windows)]
+use winapi::um::winbase::ABOVE_NORMAL_PRIORITY_CLASS;
+#[cfg(windows)]
+use winapi::um::winbase::BELOW_NORMAL_PRIORITY_CLASS;
+#[cfg(windows)]
+use winapi::um::winbase::HIGH_PRIORITY_CLASS;
+#[cfg(windows)]
+use winapi::um::winbase::IDLE_PRIORITY_CLASS;
+#[cfg(windows)]
+use winapi::um::winbase::NORMAL_PRIORITY_CLASS;
+#[cfg(windows)]
+use winapi::um::winbase::REALTIME_PRIORITY_CLASS;
+#[cfg(windows)]
+use winapi::um::winnt::PROCESS_QUERY_LIMITED_INFORMATION;
+
+#[cfg(target_os = "macos")]
+#[allow(non_camel_case_types)]
+type priority_t = i32;
+#[cfg(target_os = "linux")]
+#[allow(non_camel_case_types)]
+type priority_t = u32;
+
+#[cfg(windows)]
+pub const PRIORITY_LOW: i32 = 19;
```

Comment:
It would be useful to add a comment about where these values are from (ex. maybe a comment to this: https://github.com/libuv/libuv/blob/7b43d70be4fe9c3f9003b189e62e4f86a6a88516/include/uv.h#L1276-L1281).